lib/fabrik/VirtualHostAgent.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const Agent = require('./Agent');
+
+class VirtualHostAgent extends Agent {
+
+  constructor(settings) {
+    super({
+      json: true
+    });
+    this.settings = settings;
+  }
+
+  provision(ips, guid) {
+    const body = {
+      parameters: {}
+    };
+    return this
+      .getHost(ips, 'multi-tenancy')
+      .then(ip => this.post(ip, `tenants/${guid}/lifecycle/provision`, body, 200));
+  }
+
+  deprovision(ips, guid) {
+    const body = {
+      parameters: {}
+    };
+    return this
+      .getHost(ips, 'multi-tenancy')
+      .then(ip => this.post(ip, `tenants/${guid}/lifecycle/deprovision`, body, 200));
+  }
+
+  createCredentials(ips, guid, parameters) {
+    const body = {
+      parameters: parameters
+    };
+    return this
+      .getHost(ips, 'credentials')
+      .then(ip => this.post(ip, `tenants/${guid}/credentials/create`, body, 200));
```

Comment:
bad practice - HTTP 200 on a resource is create, calling `create` once again on URL isnt right. Can drop the suffix `create` and just have the resource as `tenants/${guid}/credentials/`

Reply:
For create credentials, I tried keeping URL format similar to what is listed here, https://cloudfoundry-incubator.github.io/service-fabrik-broker/api/agent_v1.1.html. Same is used https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/master/lib/fabrik/Agent.js#L129. 
Should I change the format?

Reply:
The ones highlighted are wrong implementation. Must not follow bad samples in existing code, when writing new code :-)
Yes please change it to the way recommended.